### PR TITLE
NUP-2464 Serialize numPredictions and inferenceType via ModelProto member of HTMPredictionModelProto and TwoGramModelProto

### LINKED ISSUE #3714: Serialize numPredictions and inferenceType via ModelProto member of HTMPredictionModelProto
Serialize numPredictions and inferenceType via ModelProto member of HTMPredictionModelProto

### PR DESCRIPTION
**Pull request by vitaly-krugl**
Fixes #3714

### CODE REVIEW

File: tests/integration/nupic/opf/htmpredictionmodel_serialization_test.py [hunk 1]
```diff
@@ -51,179 +51,182 @@
 
 
 CPP_MODEL_PARAMS = {
```
Comment (vitaly-krugl) [inline tests/integration/nupic/opf/htmpredictionmodel_serialization_test.py]: Fixed indentation below

**Comment by lscheinkman** [comment]
@vitaly-krugl  Are you refactoring all classes implementing `model.Model` or just `HTMPredictionModel`?

**Comment by vitaly-krugl** [comment]
@lscheinkman: My task concerned `HTMPredictionModel`. I'll check with @scottpurdy about the others.

**Comment by vitaly-krugl** [comment]
@lscheinkman, as discussed with @scottpurdy offline: I am going to look into refactoring `PreviousValueModel` and `TwoGramModel` along the same lines as `HTMPredictionModel`.

**Comment by vitaly-krugl** [comment]
`PreviousValueModel` doesn't have capnp serialization implemented at all, so I am only going to refactor one additional class - `TwoGramModel`

**Comment by vitaly-krugl** [comment]
Ready for review

cc @scottpurdy

File: src/nupic/frameworks/opf/model.capnp [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@0xa2163632a7294a0d;
+
+using import "/nupic/frameworks/opf/opf_utils.capnp".InferenceType;
+
+# Next ID: 5
```
Comment (scottpurdy) [inline src/nupic/frameworks/opf/model.capnp]: Fix the number here

File: src/nupic/frameworks/opf/model.py [hunk 3]
```diff
@@ -252,13 +269,28 @@ def readFromCheckpoint(cls, checkpointDir):
     model = cls.read(proto)
     return model
 
+
+  def writeBaseToProto(self, proto):
```
Comment (scottpurdy) [inline src/nupic/frameworks/opf/model.py]: this is nice

File: src/nupic/frameworks/opf/htm_prediction_model.py [hunk 4]
```diff
@@ -1335,8 +1344,8 @@ def read(cls, proto):
     model = cls(spEnable=spEnable,
                 tmEnable=tmEnable,
                 clEnable=clEnable,
-                inferenceType=inferenceType,
-                network=network)
+                network=network,
+                baseProto=proto.modelBase)
```
Comment (vitaly-krugl) [inline src/nupic/frameworks/opf/htm_prediction_model.py]: The `network` arg appears to only be passed from this `read` method in the entire `nupic` body of code.
```
~/nta/nupic $ git grep 'network='

src/nupic/frameworks/opf/htm_prediction_model.py:      network=None,
src/nupic/frameworks/opf/htm_prediction_model.py:                network=network,
vkruglikov@Vitalys-MacBook-Pro:~/nta/nupic (nup2464-htmpredmodel-serialize-missing-fields)$ 

```

File: src/nupic/frameworks/opf/model.capnp (hunk 2 above)
Comment (vitaly-krugl) [inline src/nupic/frameworks/opf/model.capnp]: @scottpurdy, I didn't notice this feedback before merging. It's fixed in my next task's changeset https://github.com/vitaly-krugl/nupic/tree/nup2463-serialize-predictedfield-and-inferenceenabled as part of NUP-2463

**Pull request merged**